client/src/components/Home/Slider/useSlider.ts

```diff
@@ -0,0 +1,15 @@
+function useSlider() {
```

Comment:
This is not a custom hook, is it?
So I am not so sure about the name. 

You should only name a function with a prefix of `use` if it is a custom hook

Reply:
ok so what I did is, for each component I did a custom hook, where all the logic would be written in, just to separate concerns, just like Vue's composable, if you are familiar with that

Reply:
ok but this can also use react hook as well, so yes it can control state as I did, you would just return the variable or function declared in it and then use it, in order to be more clear maybe I would rename it to maybe, sliderFunctions.ts

Reply:
You can decide to leave it.
I am just stating that based on standard practice.

React documentation actually said a custom hook may or may not call react hook, so this seems fine too.